src/sympc/tensor/share_tensor.py

```diff
@@ -79,15 +77,24 @@ def __init__(
         self.tensor: Optional[torch.Tensor] = None
 
         if data is not None:
+            self.data = data
```

Comment:
Q: Could we move this?
For the moment we have two ```data``` pieces that seem the same -- ```self.tensor``` and ```self.data``` and we could get one from the other using the ```encode/decode```